src/OrleansRuntime/Catalog/ActivationData.cs

```diff
@@ -321,6 +322,8 @@ public GrainId Identity
 
         public ActivationAddress Address { get; private set; }
 
+        public IServiceProvider ServiceProvider => this.ServiceScope?.ServiceProvider;
```

Comment:
Which PR was that? Maybe we should provide a util for creating grains without a silo which respects scoping